riverdriver/riverpgxv5/internal/dbsqlc/river_queue.sql

```diff
@@ -48,16 +48,14 @@ LIMIT @limit_count::integer;
 
 -- name: QueuePause :execresult
 WITH queue_to_update AS (
-    SELECT name
+    SELECT name, paused_at
     FROM river_queue
     WHERE CASE WHEN @name::text = '*' THEN true ELSE river_queue.name = @name::text END
-      AND paused_at IS NULL
     FOR UPDATE
 )
-
 UPDATE river_queue
 SET
-    paused_at = now(),
+    paused_at = coalesce(queue_to_update.paused_at, now()),
     updated_at = now()
 FROM queue_to_update
 WHERE river_queue.name = queue_to_update.name;
```

Comment:
To avoid editing a row for no reason, we could take an approach we've done in some other queries: do the update in a CTE, and then for the final return do a union w/ the updated row and original row to return only one of them.

Not a high throughput query so it doesn't matter much, but it would be a nice tweak to stay consistent in our approach.

Reply:
Okay, good call. Updated.